app/src/pages/HomePage.tsx

```diff
@@ -3,28 +3,64 @@ import React from 'react';
 import { Alignment, Box, Button, ContainingView, Direction, Image, KibaIcon, LayerContainer, PaddingSize, Spacing, Stack, Text } from '@kibalabs/ui-react';
 
 import { useAccount, useOnLinkAccountsClicked } from '../AccountContext';
+import { Dropzone } from '../components/Dropzone';
 import { ImageView } from '../components/ImageView';
+import { useGlobals } from '../globalsContext';
 import imageData from '../imageData.json';
 
+export type UpdateResult = {
+  isSuccess: boolean;
+  message: string;
+}
+
 export const HomePage = (): React.ReactElement => {
   const account = useAccount();
+  const { web3StorageClient } = useGlobals();
+  const [isUploadingImage, setIsUploadingImage] = React.useState<boolean>(false);
+  const [updatingImageResult, setUpdatingImageResult] = React.useState<UpdateResult | null>(null);
+  const shouldUseIpfs = true;
   const [stage, setStage] = React.useState<number>(1);
-  const [imageLink, setImageLink] = React.useState<string>('');
+  const [imageUrl, setImageUrl] = React.useState<string>(null);
 
   const onLinkAccountsClicked = useOnLinkAccountsClicked();
 
   const onConnectWalletClicked = async (): Promise<void> => {
     await onLinkAccountsClicked();
   };
 
-  const showImage = (imageUrl) => {
+  const onImageClicked = (tokenImageUrl: string): void => {
     setStage(2);
```

Comment:
should this be here?

Reply:
setStage is used to set the state of the screen that is showing, when the stage is 1 it shows the connect wallet button if the wallet isn't connected, when it's connected it shows the tokens available. When an image is clicked this function sets the stage to 2 which is the screen where we edit the image with frames.

Reply:
yes but setStage(2) is specified below in the onImageChosen function, why do we need it here as well?

Reply:
or is this so it goes back to stage 2 from stage 3?

Reply:
onImageChosen function is to upload to ipfs

Reply:
okay this doesnt answer the question at all but il merge and figure it out later